stock_picking_batch_creation/wizards/make_picking_batch.py

```diff
@@ -0,0 +1,468 @@
+        for device in available_devices:
+            if (
+                self._volume_condition_for_device_choice(
+                    device.min_volume,
+                    picking.volume,
+                    device.max_volume,
+                )
+                and tools.float_compare(
+                    device.max_weight,
+                    picking.weight,
+                    precision_digits=self._precision_volume(),
+                )
+                > 0
+            ):
+                return device
+        return None
+
+    def _volume_condition_for_device_choice(
+        self, min_volume, picking_volume, max_volume
+    ):
+        lower_bound = tools.float_compare(picking_volume, min_volume, 2) >= 0
+        upper_bound = tools.float_compare(max_volume, picking_volume, 2) >= 0
+        return lower_bound and upper_bound
+
+    def _raise_create_batch_not_possible(self):
+        """Raise the correct error to desrcibe why it's not possible to
+        create a batch."""
+        # At this stage we know that there is no picking to process
+        # at least for the list of devices. To provide to the user
+        # the right information about the reason why there is no picking
+        # we check if there is a candidate pickings without device
+        # constrains. If not, we raise an error to inform the user that there
+        # is no picking to process otherwise we raise an error to inform the
+        # user that there is not suitable device to process the pickings.
+        if not self.no_line_limit_if_no_candidate:
+            domain = self._get_picking_domain_for_first(no_nbr_lines_limit=True)
+            candidates = self.env["stock.picking"].search(domain, limit=1)
+            if candidates:
+                raise PickingCandidateNumberLineExceedError(
+                    candidates, self.maximum_number_of_preparation_lines
+                )
+        domain = self._get_picking_domain_for_first()
+        limit = 1
+        if self.add_picking_list_in_error:
+            limit = None
+        candidates = self.env["stock.picking"].search(domain, limit=limit)
+        if candidates:
+            pickings = candidates if self.add_picking_list_in_error else None
+            raise NoSuitableDeviceError(pickings=pickings)
+        raise NoPickingCandidateError()
+
+    def _create_batch(self, raise_if_not_possible=False):
+        """Create a batch transfer."""
+        self._reset_counters()
+        # first we try to get the first picking for the user
+        first_picking = self._get_first_picking()
+        if not first_picking and self.no_line_limit_if_no_candidate:
+            first_picking = self._get_first_picking(no_nbr_lines_limit=True)
+        if not first_picking:
+            if raise_if_not_possible:
+                self._raise_create_batch_not_possible()
+            return self.env["stock.picking.batch"].browse()
+        device = self._compute_device_to_use(first_picking)
```

Comment:
If first picking exceeds device's max_weight, create a split picking within limit.

In compute device, fallback on the last allowed device (as you will exclude them all if picking weight > all(device's max_weight)